src/werkzeug/debug/__init__.py

```diff
@@ -52,18 +52,19 @@ def get_machine_id():
         return rv
 
     def _generate():
-        # docker containers share the same machine id, get the
+        # containers share the same machine id, get the
         # container id instead
         try:
             with open("/proc/self/cgroup") as f:
                 value = f.readline()
         except IOError:
             pass
         else:
-            value = value.strip().partition("/docker/")[2]
-
-            if value:
-                return value
+            # Different container technology uses different patterns in
+            # cgroups names.
+            for pattern in ["docker", "libpod"]:
```

Comment:
I verified through manual testing that this will actually change the result returned by `get_machine_id`.

For a container with the first line of `/proc/self/cgroup` returning `13:name=systemd:/docker/ed579b6f6e96a9e0c386cf6ff88888417243a9aa39c7190034e1f42d5fd22f6b\n` the old function returned `ed579b6f6e96a9e0c386cf6ff88888417243a9aa39c7190034e1f42d5fd22f6b`. It now returns `/ed579b6f6e96a9e0c386cf6ff88888417243a9aa39c7190034e1f42d5fd22f6b`.

Updating the pattern list to `["/docker/", "/libpod/"]` will return the expected result as it restores the original intent.

Reply:
In my environment content of /proc/self/cgroup is:

11:cpuset:/system.slice/docker-60e4623a031ae369eca36a1542c1fb698db054ae0ed5e06c59724d482a7a2164.scope

Note "docker-" vs "docker/"

So it seems this really depends on distros or version in addition to container technology.